app/controllers/shopping_lists_controller.rb

```diff
@@ -23,11 +23,47 @@ def show
     head :not_found
   end
 
+  def update
+    shopping_list = current_user.shopping_lists.find(params[:id])
+
+    if shopping_list.update(shopping_list_update_params)
+      render json: shopping_list, status: :ok
+    else
+      render json: { errors: shopping_list.errors }, status: :unprocessable_entity
+    end
+  rescue ActiveRecord::RecordNotFound
+    head :not_found
+  end
+
   def destroy
     shopping_list = current_user.shopping_lists.find(params[:id])
-    shopping_list.destroy!
-    head :no_content
+
+    if shopping_list.master
```

Comment:
This will probably be able to be refactored out when I do [this card](https://trello.com/c/tPeoZBxn/3-return-error-for-non-get-methods-on-master-list-resource) since we'll be significantly restricting ability to work with master lists through the API.